HyperVExtension/src/HyperVExtension/Models/HyperVVirtualMachine.cs

```diff
@@ -451,7 +453,29 @@ public IAsyncOperation<ComputeSystemOperationResult> ConnectAsync(string options
         {
             try
             {
-                _hyperVManager.ConnectToVirtualMachine(VmId);
+                _log.Information($"Starting vmconnect launch attempt on {DateTime.Now}: VM details: {this}");
+                ProcessStartInfo processStartInfo = new ProcessStartInfo("vmconnect.exe");
+                processStartInfo.UseShellExecute = true;
+
+                // The -G flag allows us to use the Id of the VM to tell vmconnect.exe which VM to launch into.
+                processStartInfo.Arguments = $"localhost -G {Id}";
+
+                using (Process vmConnectProcess = new Process())
+                {
+                    vmConnectProcess.StartInfo = processStartInfo;
+
+                    // We start the process and will return success if it does not throw an exception.
+                    // If vmconnect has an error, it will be displayed to the user in a message dialog
+                    // outside of our control.
+                    vmConnectProcess.Start();
+
+                    // Note: Just because the vmconnect.exe launches in the foreground does not mean it will launch
+                    // in front of the Dev Home window. Since the vmconnect.exe is a separate process being launched
+                    // outside of Dev Home, it will not be parented to the Dev Home window. The shell will launch it
+                    // in its last known location.

```

Comment:
Where did you get this information? I don't see it on the MSDN page for SetForegroundWindow.

Reply:
Which part? do you mean "The shell will launch it in its last known location"? That part is not specific to `SetForegroundWindow` that is just Windows generic behavior in general as far as I know. `SetForegroundWindow` only brings it to the front. What happens after that in terms of placement is up to the shell and the win32 app I believe.